### ISSUE TITLE
代码疑惑

### ISSUE THREAD

**Issue by tongchangD** [xinntao/Real-ESRGAN#40]
```
  # gan 网络这里 将 HR 数据 传入
  # real
  real_d_pred = self.net_d(gan_gt)
  l_d_real = self.cri_gan(real_d_pred, True, is_disc=True)
  loss_dict['l_d_real'] = l_d_real
  loss_dict['out_d_real'] = torch.mean(real_d_pred.detach())
  l_d_real.backward()
  # fake
  fake_d_pred = self.net_d(self.output.detach().clone())  # clone for pt1.9
  l_d_fake = self.cri_gan(fake_d_pred, False, is_disc=True)
  loss_dict['l_d_fake'] = l_d_fake
  loss_dict['out_d_fake'] = torch.mean(fake_d_pred.detach())
  l_d_fake.backward()
  self.optimizer_d.step()
```
涛哥,这训练 RealESRGAN第二部分时, LOSS 一块的代码没有看懂,涛哥能否抽空指导指导我,请问这两部分有什么作用

**Comment by tongchangD** [comment]
对不起,我蠢了

**Comment by xinntao** [comment]
这是 训练GAN的 discriminator的代码~ 这是比较基本的~
